docs/csharp/whats-new/tutorials/primary-constructors.md

```diff
@@ -0,0 +1,110 @@
+---
+title: Declare and use primary constructors in classes and structs
+description: "Learn how and when to declare primary constructors in your class and struct types. Primary constructors provide concise syntax to declare constructor parameters available anywhere in your type."
+ms.date: 05/26/2023
+---
+# Tutorial: Explore primary constructors
+
+C# 12 introduces [*primary constructors*](../../programming-guide/classes-and-structs/instance-constructors.md#primary-constructors), a concise syntax to declare constructors whose parameters are available anywhere in the body of the type.
+
+In this tutorial, you learn how to:
+
+> [!div class="checklist"]
+>
+> - When to declare a primary constructor on your type
+> - How to call call primary constructors from other constructors
+> - How to use primary constructor parameters in members of the type
+> - Where primary constructor parameters are stored
+
+## Prerequisites
+
+You need to set up your machine to run .NET 8 or later, including the C# 12 or later compiler. The C# 12 compiler is available starting with [Visual Studio 2022 version 17.7](https://visualstudio.microsoft.com/vs) or the [.NET 8 SDK](https://dotnet.microsoft.com/download).
+
+Key learning experiences for this tutorial:
+
+1. declare a primary constructor when parameters are needed in one or more members.
+    1. initialize properties (same or different type.)
+    1. as a service or dependencies.
+    1. pass to base class with primary constructor
+1. consider when a reasonable can be used in a default ctor.
+1. Use primary ctor parameter as you would use a field.
+
+## Primary constructors
+
+You can add parameters to a `struct` or `class` declaration to create a *primary constructor*. You declare a primary constructor when a struct or class **requires** parameters to initialize an instance. Primary constructor parameters are in scope throughout the class definition. It's important to view primary constructor parameters as *parameters* even though they are in scope throughout the class definition. Several rules clarify that they're parameters:
+
+1. Primary constructor parameters aren't stored directly as fields.
```

Comment:
Instead of saying that they're not stored directly as fields (which they are in the roslyn impl), i'd say something more like "how parameters are stored isn't specified and compilers are free to store or not store them however they want".

I'd also add "primary constructor parameters may not be stored at all if not needed".